packages/manager/apps/hub/src/routing.js

```diff
@@ -30,6 +31,20 @@ export default /* @ngInject */ ($stateProvider, $urlRouterProvider) => {
             ),
           }),
         ),
+      order: /* @ngInject */ (hub, OrderTracking) =>
+        OrderTracking.getOrderStatus(hub.data.lastOrder.data)
```

Comment:
Not saying you should change anything, but this would be clearer:

```suggestion
        const orderStatus = await OrderTracking.getOrderStatus(hub.data.lastOrder.data);
        const completeHistory = await OrderTracking.getCompleteHistory(orderStatus);
        const orderDetails = await OrderTracking.getOrderDetails(order);
        
        return {
            ...order,
            ...history,
            ...head(details),
        }
```

Reply:
I agree but as we are in a `angular` context I wonder how this will integrate in the state lifecycle

Reply:
I said in my message `Not saying you should change anything`, I was just stating that it would be easier to read :)

Reply:
Also, the solution I've found to do this was

`$q
  .when()
  .then(async () => {
    const orderStatus = await OrderTracking.getOrderStatus(hub.data.lastOrder.data);
    const completeHistory = await OrderTracking.getCompleteHistory(orderStatus);

    // and so on
  })`

but seriously, it doesn't matter :)